src/server/frontend_xwayland/xwayland_wm_surface.cpp

```diff
@@ -335,3 +436,75 @@ void mf::XWaylandWMSurface::aquire_shell_surface(std::function<void(XWaylandWMSh
                 work(shell_surface);
         });
 }
+
+void mf::XWaylandWMSurface::set_window_state(WindowState const& new_window_state)
+{
+    WmState const wm_state{
+        new_window_state.minimized ?
+        WmState::ICONIC :
+        WmState::NORMAL};
+
+    uint32_t const wm_state_properties[]{
+        static_cast<uint32_t>(wm_state),
+        XCB_WINDOW_NONE // Icon window
+    };
+
+    xcb_change_property(
+        xwm->get_xcb_connection(),
+        XCB_PROP_MODE_REPLACE,
+        window, xwm->xcb_atom.wm_state,
+        xwm->xcb_atom.wm_state, 32,
+        length_of(wm_state_properties), wm_state_properties);
+
+    std::vector<xcb_atom_t> net_wm_states;
+
+    if (new_window_state.maximized)
```

Comment:
Do you mean `new_window_state.minimized` here? :confused:

Reply:
yes